networking.md

```diff
@@ -0,0 +1,92 @@
+# Filecoin Network
+
+The filecoin network is built using libp2p building blocks, as transports and protocols, as well as some additional filecoin specific protocols as outlined in [Network Protocols](./network-protocols.md).
+
+## Required Protocols
+
+Every full node must support the following libp2p protocols:
+
+ - [gossipsub](https://github.com/libp2p/specs/tree/master/pubsub/gossipsub) ([for data announcements](./data-propagation.md))
+ - [bitswap] ([for data exchange](./))
+ - Filecoin specific Protocols:
+   - Hello Handshake
+   - StorageDeal
+   - BlockSync
+
+## Transports, Streams & Encryption
+
+Connections between nodes are end-to-end-encrypted and -authenticated, thus every node has a `peerId` associated with it. Any node can also support unencrypted or unauthenticated connection request, but it is not required to do so.
+
+Filecoin protocols are run as multiple streams over a single connection. Only one connection should exist between two nodes at any time, multiple streams within that connections must be used to exchange different protocols.
```

Comment:
Maybe add a note at the end of this, "This is all abstracted away and handled by libp2p, Filecoin implementations with a correct libp2p library need not worry about this"

Reply:
I am a little hesitant to add sentences like this for two reasons: 1. with libp2p being a modular stack, it doesn't really state whether muxing is implemented, nor is it enforced by the library and 2. there is no real definition of what constitutes the "correct libp2p library"/implementation -- especially as we don't currently run cross-implementation-integration tests...

How about
```suggestion
Filecoin protocols are run as multiple streams over a single connection. Only one connection should exist between two nodes at any time, multiple streams within that connections must be used to exchange different protocols. This is usually handled from within the libp2p stack if configured properly.
```

Reply:
That sounds good to me!